pkg/metrics/metrics.go

```diff
@@ -0,0 +1,41 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+var (
+	controllerManagerEnabled = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "service_catalog_controller_manager_enabled",
+			Help: "Indicates whether Service Catalog controller manager is enabled",
+		})
+)
+
+func init() {
+	// do the MustRegister here
+	prometheus.MustRegister(controllerManagerEnabled)
+}
+
+// We will never want to panic our operator because of metric saving.
+// Therefore, we will recover our panics here and error log them
+// for later diagnosis but will never fail the operator.
+func recoverMetricPanic() {
+	if r := recover(); r != nil {
+		klog.Errorf("Recovering from metric function - %v", r)
+	}
+}
+
+// ControllerManagerEnabled - Indicates Service Catalog Controller Manager has been enabled
+func ControllerManagerEnabled() {
+	defer recoverMetricPanic()
+	controllerManagerEnabled.Inc()
```

Comment:
Since this is a gauge, it may be safer to use `.Set(1.0)` here and `.Set(0.0)` in `ControllerManagerDisabled()`. Then these functions would be idempotent.

Reply:
Switched to Set. Thanks.